make automator and provider types readable across all tenants and

albertshau opened the pull request:
writeable by the superadmin only. This is because plugins are
currently constant across all tenants, but will change once that
is no longer the case.

albertshau committed:
make automator and provider types readable across all tenants and
writeable by the superadmin only. This is because plugins are
currently constant across all tenants, but will change once that
is no longer the case.

poornachandra commented on server/src/main/java/com/continuuity/loom/http/handler/LoomAdminHandler.java:
```diff
@@ -572,8 +572,8 @@ public void deleteProviderType(HttpRequest request, HttpResponder responder,
     }
-    if (!account.isAdmin()) {
-      responder.sendError(HttpResponseStatus.FORBIDDEN, "user unauthorized, must be admin.");
+    if (!account.isSuperadmin()) {
```
Now in this class - some methods need admin authorization, and some need super-admin authorization. I think it would be better to move super-admin methods into a separate handler.

poornachandra commented on server/src/main/java/com/continuuity/loom/store/entity/BaseSQLEntityStoreView.java:
```diff
@@ -103,11 +104,24 @@
+    queryStr.append("s WHERE name=? AND ");
+    // TODO: remove once types are defined through server instead of through provisioner
+    if (entityType == EntityType.AUTOMATOR_TYPE || entityType == EntityType.PROVIDER_TYPE) {
+      queryStr.append("( tenant_id=? OR tenant_id='");
```
Can you add a comment here?

poornachandra commented on server/src/main/java/com/continuuity/loom/store/entity/BaseSQLEntityStoreView.java:
```diff
@@ -115,9 +129,21 @@ protected PreparedStatement getSelectStatement(Connection conn, EntityType entit
+    queryStr.append("s WHERE tenant_id=? ");
+    // TODO: remove once types are defined through server instead of through provisioner
+    if (entityType == EntityType.AUTOMATOR_TYPE || entityType == EntityType.PROVIDER_TYPE) {
+      queryStr.append(" OR tenant_id='");
```
Comment here too.

poornachandra commented on server/src/test/java/com/continuuity/loom/http/LoomAdminHandlerTest.java:
```diff
@@ -77,13 +78,13 @@ public void testClusterTemplates() throws Exception {
   public void testProviderTypes() throws Exception {
     testNonPostRestAPIs("providertypes", Entities.ProviderTypeExample.JOYENT_JSON,
-                        Entities.ProviderTypeExample.RACKSPACE_JSON);
+                        Entities.ProviderTypeExample.RACKSPACE_JSON, SUPERADMIN_HEADERS);
```
Can you add a negative case.

poornachandra commented:
Few comments, you can merge after fixing them :+1:

albertshau committed:
moving provider and automator type endpoints to superadmin handler
so the superadmin apis are in the same class. Also taking them
out of import/export since they are purely superadmin entities.
